libr/anal/fcn.c

```diff
@@ -265,6 +265,63 @@ static ut64 search_reg_val(RAnal *anal, ut8 *buf, ut64 len, ut64 addr, char *reg
 
 #define gotoBeach(x) ret=x;goto beach;
 #define gotoBeachRet() goto beach;
+
+R_API void fill_args (RAnal *anal, RAnalFunction *fcn, RAnalOp *op) {
+	char* varname;
+	switch (op->stackop) {
+	case R_ANAL_STACK_NULL:
+	case R_ANAL_STACK_NOP:
+	case R_ANAL_STACK_ALIGN:
+	case R_ANAL_STACK_INC: {
+		char *esil_buf = strdup ( r_strbuf_get (&op->esil));
+		char *ptr_end = strstr (esil_buf, ",ebp,+,[4],");
+		char *addr;
+		if (!ptr_end) {
+			goto end;
+		}
+		*ptr_end = 0;
+		addr = ptr_end;
+		while((*addr != '0' || *(addr+1) != 'x') &&addr >= esil_buf && *addr != ',' ) {
+			addr--;
+		}
+		if (strncmp(addr, "0x", 2)) {
+			goto end;
+		}
+		st64 ptr;
```

Comment:
why define a var in the middle of nothing and assign it in the next line? just do st64 ptr = (st64) r_num_get(.. because rnum returns ut64